README.md

```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
  </dependency>
 ```
 
-We also need to add an annotation to enable dekorate. In this case we will use `@KubernetesApplication` which also gives us access to more Kubernetes specific configuration options.
-Edit the App class and add the indicated annotation.
+We need to enable `Dekorate` by adding an annotation. In this case we will use `@KubernetesApplication` which also gives us access to more Kubernetes specific configuration options.
+Edit the App class and add the following annotation.
 
 ```
-@KubernetesApplication(name = "hello-world-fwless-k8s", ports = @Port(name = "web", containerPort = 8080))
+@KubernetesApplication(
+        name = "hello-world-fwless-k8s",        
+        ports = @Port(name = "web", containerPort = 8080),  
+        host = "fw-app.127.0.0.1.nip.io", //2
+        expose = true, //3
+        imagePullPolicy = ImagePullPolicy.Always //4
+)
 ```
-We need to specify a port in order to prevent dekorate it is about a Web Application, this way a service will be generated in the k8s manifest.
+- We need to specify a **`@Port`** in order to prevent dekorate it is about a Web Application, this way a service will be generated in the k8s manifest.
 
-NOTE: if you are using minikube a trick that often works is to use a service of type NodePort.
+- The host under which the application is going to be exposed.
+- **`expose`** controls whether the application should be exposed via Ingress.
+- We use **`Always`** in order to be able to use an updated image.
 
-Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate.
+Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate`.
 
-NOTE: To perform the following steps you need to be connected to a running kubernetes cluster.
+## Building and deploying
 
-Run the following command to create the resources defined in the manifest.
+We need to somehow build and push the image to the registry to make it available for the kubernetes cluster. 
```

Comment:
You should better introduce this part of the doc and mentions by example. `Now that we have populated YAML kubernetes resources able to deploy the application on the cluster, we must first create an container image and push it to a local container registry .... `